tools/cli/taskListCommands.go

```diff
@@ -23,22 +23,32 @@ package cli
 import (
 	"os"
 
+	cc "github.com/uber/cadence/common/client"
+
 	"github.com/uber/cadence/common/types"
 
 	"github.com/olekukonko/tablewriter"
 	"github.com/urfave/cli"
-	s "go.uber.org/cadence/.gen/go/shared"
 )
 
 // DescribeTaskList show pollers info of a given tasklist
 func DescribeTaskList(c *cli.Context) {
 	wfClient := getWorkflowClient(c)
+	domain := getRequiredGlobalOption(c, FlagDomain)
 	taskList := getRequiredOption(c, FlagTaskList)
 	taskListType := strToTaskListType(c.String(FlagTaskListType)) // default type is decision
 
 	ctx, cancel := newContext(c)
 	defer cancel()
-	response, err := wfClient.DescribeTaskList(ctx, taskList, taskListType)
+
+	request := &types.DescribeTaskListRequest{
+		Domain: domain,
+		TaskList: &types.TaskList{
+			Name: taskList,
+		},
+		TaskListType: taskListType,
+	}
+	response, err := wfClient.DescribeTaskList(ctx, request, cc.GetDefaultCLIYarpcCallOptions()...)
```

Comment:
Do we set `GetDefaultCLIYarpcCallOptions` on every call?

If so, those can be set using common middleware on the dispatcher. It would reduce such boilerplate code. But this may as well be a separate PR.

Reply:
ya, so the default yarpc options are being passed in cadence-client when `newChannelContext` is called